contracts/factories/DelegatedManagerFactory.sol

```diff
@@ -0,0 +1,378 @@
+    using Address for address;
+
+    /* ============ Structs ============ */
+
+    struct InitializeParams{
+        address deployer;
+        address owner;
+        IDelegatedManager manager;
+        bool isPending;
+    }
+
+    /* ============ Events ============ */
+
+    /**
+     * @dev Emitted on DelegatedManager creation
+     * @param _setToken             Instance of the SetToken being levered
+     * @param _manager              Address of the DelegatedManager
+     * @param _deployer             Address of the deployer
+    */
+    event DelegatedManagerCreated(
+        ISetToken indexed _setToken,
+        DelegatedManager indexed _manager,
+        address _deployer
+    );
+
+    /**
+     * @dev Emitted on DelegatedManager initialization
+     * @param _setToken             Instance of the SetToken being levered
+     * @param _manager              Address of the DelegatedManager owner
+    */
+    event DelegatedManagerInitialized(
+        ISetToken indexed _setToken,
+        IDelegatedManager indexed _manager
+    );
+
+    /* ============ State Variables ============ */
+
+    // SetTokenFactory address
+    ISetTokenCreator public setTokenFactory;
+
+    // Mapping which stores manager creation metadata between creation and initialization steps
+    mapping(ISetToken=>InitializeParams) public initializeState;
+
+    // Mapping of all sets that have succesfully initialized
+    mapping(ISetToken=>bool) public isValidSet;
+
+    // Address array of all sets that have succesfully initialized
+    address[] internal validSets;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * @dev Sets setTokenFactory address.
+     * @param _setTokenFactory                  Address of SetTokenFactory protocol contract
+     */
+    constructor(ISetTokenCreator _setTokenFactory) public {
+        setTokenFactory = _setTokenFactory;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ANYONE CAN CALL: Deploys a new SetToken and DelegatedManager. Sets some temporary metadata about
+     * the deployment which will be read during a subsequent intialization step which wires everything
+     * together.
+     *
+     * @param _components       List of addresses of components for initial Positions
+     * @param _units            List of units. Each unit is the # of components per 10^18 of a SetToken
+     * @param _name             Name of the SetToken
+     * @param _symbol           Symbol of the SetToken
+     * @param _owner            Address to set as the DelegateManager's `owner` role
+     * @param _methodologist    Address to set as the DelegateManager's methodologist role
+     * @param _modules          List of modules to enable. All modules must be approved by the Controller
+     * @param _operators        List of operators authorized for the DelegateManager
+     * @param _assets           List of assets DelegateManager can trade. When empty, manager can trade any asset
```

Comment:
instead of `can trade any asset` maybe a clearer way to state would be `asset allow list is not enforced`

Reply:
[c2540a6](https://github.com/SetProtocol/set-v2-strategies/pull/14/commits/c2540a600a374b60e669d04c332498a3a587dda8)